reader.go

```diff
@@ -261,3 +261,24 @@ func (r *Reader) WriteTo(w io.Writer) (n int64, err error) {
 		}
 	}
 }
+
+// CheckLZ4 returns a bool indicating if the first 32 bytes mactch LZ4 header
+// from the io.Reader stream.
+//
+// False if it could not determine, e.g. error in reading from io.Reader.
+//
+// Don't pass original io.Reader for checking, otherwise internal read pos of
+// the io.Reader will be off by 32 bytes, causing a reader error. Instead, consider
```

Comment:
this is a really awkward thing that will result in people setting off this footgun. suggest this gets enforced a different way, because folks won't necessarily read this comment and get surprised when it type checks and yet they can't read a "valid" file that they verified

Reply:
how would you like it done differently ?

even we handle it differently, say, take a []bytes as input, and match for lz4 header, users still need to **read** it, and pass it in. Whenever they read it, the reading offset of the io.Reader will increment.

Reply:
Why not return a MultiReader as you discuss, if it is a valid LZ4 stream?

Reply:
This can be changed to take a `[]byte` if that really bugs you. I think at lz4 package level, it should stay simple, and caller should worry about those things. For an example, the io.Reader can be a `bytes.NewReader([]byte)`, where the bugs is a pre-read slice from their original `io.Reader`, having lz4 return a Multi-reader is by assuming user is passing in its original io.Reader ? that assumption not a good idea, IMO

Reply:
I would definitely assume that it was safe to pass in the original Reader, but maybe that is just me.

Reply:
What about returning the read byte[] that otherwise might get eaten from the stream? Then caller can either discard it or can use it to construct the MultiReader

Reply:
Done.

changed it to take `[]byte` instead. Caller need to handle getting the bytes however it likes. `lz4` might as well be agnostic to all those confusion, and multireader bussiness.